### PR TITLE
Bug fix/forbid id index creation

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] The behaviour change can only be verified via automatic tests

#### Related Information
- [ ] There is a GitHub Issue reported by a Community User: https://github.com/arangodb/arangodb/issues/9118
- [ ] There is an internal planning ticket: https://github.com/arangodb/team-orco-planning/issues/136

### Testing & Verification
Upgrading older databases will simply not instanciate this index on startup.

this will be permitted by the old arangodb:
`db._users.ensureIndex({type: "hash", "fields": ["somefield._id"] }) ` 
when running the fixed version over an existing database an error will be logged and re-creation of the index will be blocked:
```
2019-06-21T11:47:46Z [87576] ERROR [5384d] {engines} failed to instantiate index, Index definition wasn't valid - factory returned null instance
2019-06-21T11:47:46Z [87576] ERROR [ddafd] {engines} error creating index from definition '{\n  "id" : "169",\n  "type" : "hash",\n  "name" : "idx_1636679789968359424",\n  "fields" : [\n    "other._id"\n  ],\n  "objectId" : "168",\n  "unique" : false,\n  "sparse" : false,\n  "deduplicate" : true\n}'
```

This PR adds tests that were used to verify all changes:
- [x] Added new **integration tests** js/common/shell/shell-index-ensure.js

### Documentation

- [x] Added a *Changelog Entry*

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4925/

File: arangod/Indexes/IndexFactory.cpp [hunk 1]
```diff
@@ -339,6 +350,15 @@ TRI_idx_iid_t IndexFactory::validateSlice(arangodb::velocypack::Slice info,
   return iid;
 }
 
+namespace {
+bool lastFieldIs(arangodb::velocypack::StringRef const& field, std::string const& match) {
```
Comment (ObiWahn) [inline arangod/Indexes/IndexFactory.cpp]: This the final name is, @dothebart?

**Comment by graetzer** [comment]
probably better to generate an error ?

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4930/ - tests blue

**Comment by jsteemann** [comment]
Superseded by #9337

**Pull request closed without merging**